zebra-rpc/src/methods.rs

```diff
@@ -518,6 +537,88 @@ where
         }
         .boxed()
     }
+
+    fn z_get_treestate(&self, hash_or_height: String) -> BoxFuture<Result<GetTreestate>> {
+        let mut state = self.state.clone();
+
+        async move {
+            let hash_or_height = hash_or_height
+                .parse()
+                .map_err(|error: SerializationError| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+
+            let block_request = zebra_state::ReadRequest::Block(hash_or_height);
```

Comment:
Performance note: reading the whole block could be expensive, we could just read the block header. But this doesn't matter unless `z_get_treestate` gets called a lot.

```suggestion
            // TODO: if this RPC is called a lot, just get the block header, rather than the whole block
            let block_request = zebra_state::ReadRequest::Block(hash_or_height);
```

Reply:
That's a good point; however, the RPC response needs to contain block hash and height. In order to get these two values, I'm calling `block.hash()` and `block.coinbase_height()` in the newer commits. I'm not sure if these two methods are suitable or if the whole approach could be improved. I'm assuming the block hash computation is fast and cheap.

Reply:
It seems fine for now.

It's possible to just get the block header and hash it, then get the coinbase transaction, and get the height. But that's more complicated, so it's not worth doing unless this RPC is called a lot.